app/src/main/contrib/sources/ParametersTokenSource.ts

```diff
@@ -0,0 +1,26 @@
+import SourceModel = AltJS.SourceModel;
+import {settings} from "../../shared/Settings";
+import {modelParameterActions} from "../actions/ModelParameterActions";
+import {ResponsibilityState} from "../stores/ResponsibilityStore";
+import {Source} from "../../shared/sources/Source";
+
+export class ModelParametersTokenSource extends Source<ResponsibilityState> {
+
+    _fetchOneTimeParametersToken: () => SourceModel<string>;
+
+    constructor() {
+        super();
+        this._fetchOneTimeParametersToken = () => this.doFetch(state => {
+            let queryString = "";
+            if (state.redirectPath && state.redirectPath.length > 0) {
+                queryString = "?redirectUrl=" + encodeURI(settings.montaguUrl() + state.redirectPath);
+            }
+            return `/modelling-groups/${state.currentModellingGroup.id}/model-run-parameters/`
+                + `${state.currentTouchstone.id}/get_onetime_link/${queryString}`;
+        }, {
```

Comment:
Something weird with the indentation here

Reply:
how would you format this? this is how my IDE wants to do it...

Reply:
Oh, I realise now that I was misreading the code. But that makes me think even more that this isn't the clearest layout. How about [like this](https://github.com/vimc/montagu-webapps/commit/0d7833d7ed4b3d92df48b2a43c221ae47b6313e5)